lib/src/realm_class.dart

```diff
@@ -107,12 +116,8 @@ class Realm {
   }
 
   void _populateMetadata() {
-    for (var realmClass in config.schema) {
-      final classMeta = realmCore.getClassMetadata(this, realmClass.name, realmClass.type);
-      final propertyMeta = realmCore.getPropertyMetadata(this, classMeta.key);
-      final metadata = RealmMetadata(classMeta, propertyMeta);
-      _metadata[realmClass.type] = metadata;
-    }
+    schema = config.schemaObjects.isNotEmpty ? RealmSchema(config.schemaObjects) : realmCore.readSchema(this);
```

Comment:
What will happen if we create a Realm without schema we open the realm for the first time. The realm file will be created empty? I saw that this throw is removed `throw RealmError("No schema specified");`.

Reply:
Nothing - it will be an empty file.

Reply:
OK